src/ert/parsing/lark_parser_error_info.py

```diff
@@ -0,0 +1,83 @@
+from dataclasses import InitVar, dataclass
+from typing import List, Optional, Union
+
+from .lark_parser_file_context_token import FileContextToken
+from .lark_parser_types import MaybeWithContext
+
+
+@dataclass
+# pylint: disable=too-many-instance-attributes
+class ErrorInfo:
+    message: str
+    filename: Optional[str]
+    start_pos: Optional[int] = None
+    line: Optional[int] = None
+    column: Optional[int] = None
+    end_line: Optional[int] = None
+    end_column: Optional[int] = None
+    end_pos: Optional[int] = None
+    originates_from: InitVar[Union[MaybeWithContext, FileContextToken]] = None
```

Comment:
Should name `Union[MaybeWithContext, FileContextToken]` as something. Possibly use a `Protocol`.

Reply:
Resolved by adding `FileContextToken` to `MaybeWithContext` type